// Copyright 2017 The CrunchyCrypt Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package crunchy;
option java_package = "com.google.security.crunchy.internal.keys";

option java_outer_classname = "SigningKeyTestVectorsOuterClass";
option java_multiple_files = true;

import "crunchy/key_management/internal/keyset.proto";

message SigningKeyTestVector {
  KeyData private_key_data = 1;
  KeyData public_key_data = 2;
  bytes message = 3;
  bytes signature = 4;
}

message SigningKeyTestVectors {
  repeated SigningKeyTestVector test_vector = 1;
}
